### PR TITLE
 	Add task Building & Packaging Web Apps 

### PR DESCRIPTION
**Pull request by IvelinYR**

### CODE REVIEW

File: Build with webpack AppBikes/.idea/modules.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mgenov) [inline Build with webpack AppBikes/.idea/modules.xml]: .idea project folder need to be added to .gitignore

File: html css javascript/form/main.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+function validateForm() {
```
Comment (mgenov) [inline html css javascript/form/main.js]: This code should not be part of this PR.

File: html css javascript/form/index.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline html css javascript/form/index.html]: This code should not be part of this PR.

File: html css javascript/calculator/style.css [hunk 4]
```diff
@@ -0,0 +1,69 @@
+body{
```
Comment (mgenov) [inline html css javascript/calculator/style.css]: This code should not be part of this PR.

File: html css javascript/calculator/main.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+function validateSign(value) {
```
Comment (mgenov) [inline html css javascript/calculator/main.js]: This code should not be part of this PR.

File: html css javascript/calculator/index.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline html css javascript/calculator/index.html]: This code should not be part of this PR.

File: Build with webpack AppBikes/src/footer.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+
+import './footer.css'
+import './clearfix.css'
+const Footer = React.createClass({
```
Comment (mgenov) [inline Build with webpack AppBikes/src/footer.js]: You can make it shorter with: 

```javascript
const Footer = () => (
   <div className='footer-box u-clearfix'>
      ...
   </div>
)
```

File: Build with webpack AppBikes/src/index.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="app.styles.css">
+    <style>
+      #root{
```
Comment (mgenov) [inline Build with webpack AppBikes/src/index.html]: It's strange some of the CSS to be in app.styles.css and some of them to be in the index page.

File: Build with webpack AppBikes/.idea/modules.xml (hunk 1 above)
Comment (IvelinYR) [inline Build with webpack AppBikes/.idea/modules.xml]: I will add .idea in .gitignore

File: html css javascript/form/main.js (hunk 2 above)
Comment (IvelinYR) [inline html css javascript/form/main.js]: Sorry this is my mistake, I  forgot to  make pull request a week ago.

File: Build with webpack AppBikes/src/footer.js (hunk 7 above)
Comment (IvelinYR) [inline Build with webpack AppBikes/src/footer.js]: I fixed it. Thank you I had not seen this syntax. It's very nice.

File: Build with webpack AppBikes/src/index.html (hunk 8 above)
Comment (IvelinYR) [inline Build with webpack AppBikes/src/index.html]: I forgot to put this code in another file. I fixed it.

**Comment by IvelinYR** [comment]
I made all the changes.

**Comment by mgenov** [comment]
Doesn't look so. I'll add my comments inline.

File: html css javascript/calculator/index.html (hunk 6 above)
Comment (mgenov) [inline html css javascript/calculator/index.html]: https://www.google.bg/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=git+committed+.gitignore+file&*
http://stackoverflow.com/questions/7527982/applying-gitignore-to-committed-files

File: html css javascript/form/main.js (hunk 2 above)
Comment (mgenov) [inline html css javascript/form/main.js]: You can make separate PR for this task only. 
```
cp -fr "html css javascript" -R ../temp 
git checkout master
git checkout -b html_js_css_task
mv ../temp html_css_javascript
git add .
git commit 
git push origin html_js_css_task
```

You choose better folder names, but this should be easy enough for fixing.

File: Build with webpack AppBikes/.idea/modules.xml (hunk 1 above)
Comment (mgenov) [inline Build with webpack AppBikes/.idea/modules.xml]: It's added but idea project files are still committed

https://www.google.bg/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=git+committed+.gitignore+file&*
http://stackoverflow.com/questions/7527982/applying-gitignore-to-committed-files

If you delete the .idea folder then you have to re-import project folder in IDEA.

Comment (IvelinYR) [inline Build with webpack AppBikes/.idea/modules.xml]: Ok I think  understand you. Tomorrow I will try to fix it.
